tools/SourceKit/lib/SwiftLang/SwiftSourceDocInfo.cpp

```diff
@@ -853,6 +855,60 @@ static bool passCursorInfoForDecl(const ValueDecl *VD,
   return false;
 }
 
+static clang::DeclarationName
+getClangDeclarationName(clang::ASTContext &Ctx, NameTranslatingInfo &Info) {
+  assert(SwiftLangSupport::getNameKindForUID(Info.NameKind) == NameKind::ObjC);
+  if (Info.BaseName.empty() == Info.ArgNames.empty()) {
+    // cannot have both.
+    return clang::DeclarationName();
+  }
+  if (!Info.BaseName.empty()) {
+    return clang::DeclarationName(&Ctx.Idents.get(Info.BaseName));
+  } else {
+    ArrayRef<StringRef> Args = llvm::makeArrayRef(Info.ArgNames);
+    std::vector<clang::IdentifierInfo *> Pieces(Args.size());
+    std::transform(Args.begin(), Args.end(), Pieces.begin(),
+      [&](StringRef T) { return &Ctx.Idents.get(T); });
+    return clang::DeclarationName(Ctx.Selectors.getSelector(
+      /*Calculate Args*/std::accumulate(Args.begin(), Args.end(), (unsigned)0,
+        [](unsigned I, StringRef T) { return I + (T.endswith(":") ? 1 : 0); }),
```

Comment:
This is not correct; the colon is not part of the IdentifierInfo for a selector piece. You'll need to actually distinguish between zero-arg and one-arg selectors explicitly in your API, or chop off the colons before making IdentifierInfos out of them.